62_CAD/DrawResourcesFiller.cpp

```diff
@@ -2649,4 +2684,256 @@ void DrawResourcesFiller::flushDrawObjects()
 		drawCalls.push_back(drawCall);
 		drawObjectsFlushedToDrawCalls = resourcesCollection.drawObjects.getCount();
 	}
+}
+
+DrawResourcesFiller::TileUploadData DrawResourcesFiller::generateTileUploadData(const ImageType imageType, const float64_t3x3& NDCToWorld, GeoreferencedImageStreamingState* imageStreamingState)
+{
+	// I think eventually it's better to just transform georeferenced images that aren't big enough into static images and forget about them
+	if (imageType == ImageType::GEOREFERENCED_FULL_RESOLUTION) //Pass imageID as parameter, down from the addGeoRef call
+		return TileUploadData{ {}, imageStreamingState->georeferencedImageParams.worldspaceOBB };
+
+	// Compute the mip level and tile range we would need to encompass the viewport
+	GeoreferencedImageTileRange viewportTileRange = computeViewportTileRange(NDCToWorld, imageStreamingState);
+
+	// A base mip level of x in the current mapped region means we can handle the viewport having mip level y, with x <= y < x + 1.0
+	// without needing to remap the region. When the user starts zooming in or out and the mip level of the viewport falls outside this range, we have to remap
+	// the mapped region.
+	const bool mipBoundaryCrossed = viewportTileRange.baseMipLevel != imageStreamingState->currentMappedRegion.baseMipLevel;
+
+	// If we moved a huge amount in any direction, no tiles will remain resident, so we simply reset state
+	const bool relativeShiftTooBig = nbl::hlsl::any
```

Comment:
add "this is considering no mip boundary crossing" to the comments of this, because it should be only evaluated when no mip boundary was crossed